csrc/index_compute.cpp

```diff
@@ -3171,4 +3171,85 @@ Val* Index::eye(
   return GpuLower::current()->commonScalarMap().hoistScalar(result, loops);
 }
 
+Val* Index::cpAsyncBulkIndex(
+    TensorView* tv,
+    const std::vector<kir::ForLoop*>& loops) {
+  using namespace tma;
+
+  TORCH_INTERNAL_ASSERT(
```

Comment:
Side note: @liqiangxl mentioned in one of comments that we should use `NVF_CHECK` and `NVF_ERROR` instead of `TORCH_INTERNAL_ASSERT`.

Reply:
fixed